cmd/traffic/cmd/manager/internal/state/intercept.go

```diff
@@ -60,9 +60,13 @@ func (s *State) PrepareIntercept(ctx context.Context, cr *managerrpc.CreateInter
 		return &managerrpc.PreparedIntercept{Error: err.Error(), ErrorCategory: int32(errcat.GetCategory(err))}, nil
 	}
 
+	if s.config.Mode.IsTeam() && cr.ApiKey == "" {
+		return interceptError(errcat.User.New("A login is required to intercept while in Team mode."))
+	}
+
 	env := managerutil.GetEnv(ctx)
-	if env.InterceptDisableGlobal {
-		if cr.InterceptSpec.Mechanism != "http" {
+	if cr.InterceptSpec.Mechanism != "http" {
+		if env.InterceptDisableGlobal {
```

Comment:
I don't understand why this change is needed. If anything, it's more efficient to test the env first (it's just one indirection).